mne/decoding/search_light.py

```diff
@@ -669,3 +703,22 @@ def _fix_auc(scoring, y):
                                  'two-class problems.')
             y = LabelEncoder().fit_transform(y)
     return y
+
+
+def _create_progressbar_context(inst, X, message):
+    if _check_verbose(inst.verbose):
+        multiply = (len(inst.estimators_)
+                    if isinstance(inst, GeneralizingEstimator) else 1)
+        n_steps = X.shape[-1] * max(1, multiply)
+        mesg = f'{message} {inst.__class__.__name__}'
+        context = ProgressBar(
+            n_steps, mesg=mesg, position=inst.position)
+    else:
+        context = contextlib.nullcontext()
```

Comment:
Rather than `nullcontext` couldn't you do
```
which_tqdm = 'off' if not _check_verbose(inst.verbose) else 'auto'
...
context = ProgressBar(..., which_tqdm=which_tqdm)
```
?
This would avoid the need for `if pb is not None` etc. I think

Reply:
ok, I'll check.

Reply:
@larsoner I've added `which_tqdm` argument to `Progressbar` (previously it only used environment variable for setting tqdm). It should be better now. :)